flink-runtime/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaRpcService.java

```diff
@@ -197,11 +201,11 @@ public void stopService() {
 	}
 
 	@Override
-	public String getAddress(RpcGateway selfGateway) {
+	public String getAddress(RpcGateway gateway) {
 		checkState(!stopped, "RpcService is stopped");
 
-		if (selfGateway instanceof AkkaGateway) {
-			ActorRef actorRef = ((AkkaGateway) selfGateway).getRpcEndpoint();
+		if (gateway instanceof AkkaGateway) {
+			ActorRef actorRef = ((AkkaGateway) gateway).getActorRef();
 			return AkkaUtils.getAkkaURL(actorSystem, actorRef);
```

Comment:
yeah, you are right. It is a problem I have missed checking. Added the connected actor to the caches in case of mixed actor system to make current solution work